### PR TITLE
unify prod/dev add-on build

### PR DESCRIPTION
**Pull request by stefanpenner**
@rwjblue / @fivetanley r?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -22,7 +27,16 @@ module.exports = {
 
   init: function() {
     var bowerDeps = this.project.bowerDependencies();
+
     var VersionChecker = require('ember-cli-version-checker');
+    var options = this.options = this.options || {};
```
Comment (pangratz) [inline index.js]: Is `this.options` intentionally assigned to `this.options || {}` or is this a "typo"?

Comment (stefanpenner) [inline index.js]: intentional, it aims to default it to {} if it is not already set

Comment (pangratz) [inline index.js]: But it's never used again in this file, that's why I am asking. Is `init` called multiple times?

Comment (stefanpenner) [inline index.js]: Well, an ancestor could set it, and a downstream could use it.

Comment (pangratz) [inline index.js]: Ah, alright. Thanks for the clarification :gift:

**Comment by stefanpenner** [comment]
the failure appears to be:

```js
 1) Acceptance: generate and destroy adapter blueprints adapter throws when --base-class is same as name:
     Error: the object {
  "errorLog": [
    {
      "isSilentError": true
      "message": "Adapters cannot extend from themself. To resolve this, remove the `--base-class` option or change to a different base-class."
      "name": "SilentError"
      "suppressedStacktrace": true
    }
  ]
  "exitCode": 1
  "inputStream": []
  "outputStream": [
    "version: 2.4.1\n"
    "\u001b[33mThe option '--disable-analytics' is not registered with the generate command. Run `ember generate --help` for a list of supported options.\u001b[39m\n"
    "\u001b[33mThe option '--watcher' is not registered with the generate command. Run `ember generate --help` for a list of supported options.\u001b[39m\n"
    "\u001b[33mThe option '--skipGit' is not registered with the generate command. Run `ember generate --help` for a list of supported options.\u001b[39m\n"
    "installing adapter\n"
  ]
  "statusCode": 1
} was thrown, throw an Error :)
```

**Comment by stefanpenner** [comment]
ah, it might be ember-cli bump

**Comment by stefanpenner** [comment]
Alright, this should be good for review.

**Comment by stefanpenner** [comment]
r?

File: index.js (hunk 1 above)
Comment (rwjblue) [inline index.js]: Also, it is used just below (via `options.babel` which is also the same object as `this.options.babel`).

**Comment by rwjblue** [comment]
This looks good to me, but I need pull it down and test:

* linked app in production
* linked app in development
* app with globals output 

Will try to do it tonight/tomorrow morning...

**Comment by bmac** [comment]
Code looks good. I'll test out the 3 scenarios @rwjblue mentioned later today.

File: index.js [hunk 2]
```diff
@@ -22,7 +27,16 @@ module.exports = {
...
+
+    options.babel = options.babel || {};
+
+    if (process.env.EMBER_ENV === 'production') {
+      add(options.babel, 'backlist', ['es6.modules', 'useStrict']);
```
Comment (bmac) [inline index.js]: `backlist` -> `blacklist`

File: index.js [hunk 3]
```diff
@@ -22,7 +27,16 @@ module.exports = {
 
   init: function() {
     var bowerDeps = this.project.bowerDependencies();
+
     var VersionChecker = require('ember-cli-version-checker');
+    var options = this.options = this.options || {};
+
+    options.babel = options.babel || {};
+
+    if (process.env.EMBER_ENV === 'production') {
```
Comment (bmac) [inline index.js]: Does this get set automatically by an `ember build --environment=production`? When testing locally I had to set it manually.

Comment (rwjblue) [inline index.js]: Yes, it does get set automatically, but not before `init` of addons.  It is set here: https://github.com/ember-cli/ember-cli/blob/3073a5c6a67debc6f4a1c37dc404ccab126d7309/lib/models/builder.js#L45 during the starting of the build.

**Pull request closed without merging**